src/server/v2.0/handler/scanexport.go

```diff
@@ -0,0 +1,237 @@
+	// is available for all projects
+	// TODO : Should we just ignore projects that do not have the required level of access?
+
+	projects := params.Criteria.Projects
+	for _, project := range projects {
+		if err := se.RequireProjectAccess(ctx, project, rbac.ActionCreate, rbac.ResourceScan); err != nil {
+			return se.SendError(ctx, err)
+		}
+	}
+
+	scanDataExportJob := new(models.ScanDataExportJob)
+
+	secContext, err := se.GetSecurityContext(ctx)
+
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+
+	// vendor id associated with the job == the user id
+	usr, err := se.userMgr.GetByName(ctx, secContext.GetUsername())
+
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+
+	if usr == nil {
+		error := &models.Error{Message: fmt.Sprintf("User : %s not found", secContext.GetUsername())}
+		errors := &models.Errors{Errors: []*models.Error{error}}
+		return operation.NewExportScanDataForbidden().WithPayload(errors)
+	}
+
+	userContext := context.WithValue(ctx, export.CsvJobVendorIDKey, usr.UserID)
+
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+
+	jobID, err := se.scanDataExportCtl.Start(userContext, se.convertToCriteria(params.Criteria, secContext.GetUsername(), usr.UserID))
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	scanDataExportJob.ID = jobID
+	return operation.NewExportScanDataOK().WithPayload(scanDataExportJob)
+}
+
+func (se *scanDataExportAPI) GetScanDataExportExecution(ctx context.Context, params operation.GetScanDataExportExecutionParams) middleware.Responder {
+	err := se.RequireAuthenticated(ctx)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	execution, err := se.scanDataExportCtl.GetExecution(ctx, params.ExecutionID)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	sdeExec := models.ScanDataExportExecution{
+		EndTime:     strfmt.DateTime(execution.EndTime),
+		ID:          execution.ID,
+		StartTime:   strfmt.DateTime(execution.StartTime),
+		Status:      execution.Status,
+		StatusText:  execution.StatusMessage,
+		Trigger:     execution.Trigger,
+		UserID:      execution.UserID,
+		JobName:     execution.JobName,
+		UserName:    execution.UserName,
+		FilePresent: execution.FilePresent,
+	}
+
+	return operation.NewGetScanDataExportExecutionOK().WithPayload(&sdeExec)
+}
+
+func (se *scanDataExportAPI) DownloadScanData(ctx context.Context, params operation.DownloadScanDataParams) middleware.Responder {
+	err := se.RequireAuthenticated(ctx)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+	execution, err := se.scanDataExportCtl.GetExecution(ctx, params.ExecutionID)
+	if err != nil {
+		se.SendError(ctx, err)
+	}
+
+	// check if the execution being downloaded is owned by the current user
+	secContext, err := se.GetSecurityContext(ctx)
+	if err != nil {
+		return se.SendError(ctx, err)
+	}
+
+	if secContext.GetUsername() != execution.UserName {
+		return middleware.ResponderFunc(func(writer http.ResponseWriter, producer runtime.Producer) {
```

Comment:
here we can just return an forbidden error?

Reply:
Ack. Will pick this in a subsequent CL

Reply:
Done